include/login_functions.php

```diff
@@ -288,23 +313,29 @@ function auth($SessionCachePolicy = 'private_no_expire')
         return authenticate($login, $passwd, $SessionCachePolicy, isset($_POST['rememberme']));
     } else {                                         // arrive from session var
         $cookiename = str_replace('.', '_', 'CDash-' . $_SERVER['SERVER_NAME']); // php doesn't like dot in cookie names
+        $service = ServiceContainer::getInstance();
+        /** @var Session $session */
+        $session = $service->get(Session::class);
+
         if (isset($_COOKIE[$cookiename])) {
             $cookievalue = $_COOKIE[$cookiename];
-            $cookiekey = substr($cookievalue, strlen($cookievalue) - 33);
+            $cookiekey = substr($cookievalue, strlen($cookievalue) - 32);
```

Comment:
Curious, what led to this change of 33 to 32?

Reply:
The call to `generate_password(32)`.